doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,252 @@
+
+In doing so, we are counting the number of positions where the bottom right
+corner of the kernel can be placed, which is equivalent to ignoring the upmost
+:math:`k - 1` rows and the leftmost :math:`k - 1` columns of the input.
+
+From this, we conclude that the output size :math:`o` of the convolution is
+
+.. math::
+
+    o = i - (k - 1).
+
+Zero padding
+============
+
+If we factor in zero padding, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s = 1`,
+* padding size :math:`p`,
+
+all we need to do is to apply the previous rule on the augmented input. Since
+padding with :math:`p` zeros adds :math:`2p` to the input size (:math:`p` on
+each side), the output size :math:`o` of the convolution becomes
+
+.. math::
+
+    o = i + 2p - (k - 1).
+
+Here is an example for :math:`i = 4`, :math:`k = 4` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/arbitrary_padding_no_strides.gif
+
+Zero padding: special cases
+===========================
+
+There are two special cases of zero padding to take note of, because they're
+quite useful in practice.
+
+Same
+----
+
+If :math:`k` is odd (:math:`k = 2n + 1`) and
+:math:`p = \lfloor k / 2 \rfloor = n`, we have that
+
+.. math::
+
+    o = i + 2 \lfloor k / 2 \rfloor - (k - 1)
+      = i + 2n - 2n
+      = i.
+
+In other words, the output size is equal to the input size. This is sometimes
+referred to as ``'same'`` padding. Here is an example for :math:`i = 3`,
+:math:`k = 3` and :math:`p = 1`:
+
+.. figure:: conv_arithmetic_figures/same_padding_no_strides.gif
+
+Full
+----
+
+If :math:`p = k - 1`, we have that
+
+.. math::
+
+    o = i + 2(k - 1) - (k - 1)
+      = i + (k - 1).
+
+The input size is *increased* by :math:`k - 1` rather than decreased by
+:math:`k - 1`. This is sometimes referred to as ``'full'`` padding. Here is
+an example for :math:`i = 3`, :math:`k = 3` and :math:`p = 2`:
+
+.. figure:: conv_arithmetic_figures/full_padding_no_strides.gif
+
+Strides
+=======
+
+Up until now, we only considered convolutions where the kernel was translated by
+increments of :math:`s = 1`. Let's now examine what happens when :math:`s > 1`.
+To keep things simple, let's ignore padding, i.e.
+
+* input size :math:`i`,
+* kernel size :math:`k`,
+* stride :math:`s`,
+* padding size :math:`p = 0`.
+
+Here is what the strided convolution looks like for :math:`i = 5`, :math:`k = 3`
+and :math:`s = 2`:
+
+.. figure:: conv_arithmetic_figures/no_padding_strides.gif
+
+Once again, we are counting the number of positions where the bottom right
+corner of the kernel can be placed. Starting from the bottom right corner of
+a kernel placed at the top left corner of the input, we count the number of
+:math:`s`-sized hops we can make. This is equivalent to counting the number
+of groups of :math:`s` rows and columns we can make when ignoring the first
+:math:`k` rows and columns. We then need to add 1, because this counting method
+ignores the bottom right corner of the kernel.
```

Comment:
Should you change the other description, make this sentences reflect the same changes as well. 
In both cases, I think that the last sentence is difficult to comprehend and should be reformulated.